server/src/server.js

```diff
@@ -645,6 +645,14 @@ app.get("/images/:imageid", function (req, res) {
         anonAnalytics.pageview(analyticsUrl).send();
       }
       res.header("Content-Type", obj.contentType);
+      if (req.query.filename) {
+        let extension = "." + obj.contentType.split("/").pop();
+        let filename= req.query.filename;
+        if (! filename.endsWith(extension)) {
+          filename = filename + extension;
+        }
+        res.header("Content-Disposition", `attachment; filename="${filename}"`)
+      }
```

Comment:
Can we sign the filename with a hash that uses a salt that only the server knows? The shot resource knows about the shot creation date, and the image resource knows about the content type (and thus the file extension), but neither knows enough to construct the entire filename by itself.